src/Generator.Bind/EnumProcessor.cs

```diff
@@ -56,15 +56,15 @@ public EnumProcessor(IBind generator, IEnumerable<string> overrides)
             Overrides = overrides;
         }
 
-        public EnumCollection Process(EnumCollection enums, string apiname)
+        public EnumCollection Process(EnumCollection enums, string apiname, HashSet<string> extensions)
```

Comment:
This is a fair comment, though I don't think it's that big of a concern in this code. This code already has a style of very mutable state, and doing anything that doesn't match the current style would draw a lot of attention to it as special for some reason.
So making this `IReadOnlyCollection<string>` would make it safer for modification, but it would make the readability of the code harder as one would wonder why it's a `IReadOnlyCollection<string>` and not just something like `List<string>`.

Basically I think it's not worth it to change that for this code that is basically at the end of it's maintenance. We are going to retire this code with OpenTK 5.

Reply:
Then consider `IReadOnlySet<string>`.  When in doubt, prefer to pass around immutable data; it makes the _correctness_ of the code far easier to demonstrate.  Knowing that `Contains()` runs in O(1) time is useful, but being certain that a callee can't modify the data allows everything that works with that data to be more easily refactored and modified in the future.  Slow code that works with immutable data can readily be made faster; but fast code that works with mutable data can often be hard or even impossible to modify, no matter how good it is.

Reply:
And even if this code is being retired for OpenTK 5, that still does not justify writing it less maintainably.  Even after OpenTK 5 is released, OpenTK 4 will still exist, and we may even need to fix bugs in it and release patched versions.  We owe it to our future selves to make this as maintainable as possible.